src/Microsoft.OData.Core/ODataBatchReader.cs

```diff
@@ -108,12 +101,62 @@ public ODataBatchReaderState State
                 return this.batchReaderState;
             }
 
-            private set
+            set
             {
                 this.batchReaderState = value;
             }
         }
 
+        /// <summary>
+        /// >The batch-specific URL converter that stores the content IDs found in a changeset and supports resolving cross-referencing URLs.
+        /// </summary>
+        internal ODataBatchPayloadUriConverter PayloadUriConverter
+        {
+            get { return this.payloadUriConverter; }
+        }
+
+        /// <summary>
+        /// Reader's input context.
+        /// </summary>
+        protected ODataInputContext InputContext
+        {
+            get { return this.inputContext; }
+        }
+
+        /// <summary>
+        /// Previously cache contentId that should be applied to the next message read.
+        /// </summary>
+        protected string ContentIdToAddOnNextRead
+        {
+            get { return this.contentIdToAddOnNextRead; }
+            set { this.contentIdToAddOnNextRead = value; }
+        }
+
+        /// <summary>
+        /// The reader's Operation state
+        /// </summary>
+        protected OperationState ReaderOperationState
+        {
+            get { return this.operationState; }
+            set { this.operationState = value; }
+        }
+
+        /// <summary>
+        /// Boolean flag indicating whether legacy content Id behavior is used.
+        /// </summary>
+        protected bool AllowLegacyContentIdBehavior

```

Comment:
>AllowLegacyContentIdBehavior [](start = 23, length = 28)

I think this is specific to the multipart/mixed format, and shouldn't be in the base class.

Reply:
removed.

---
In reply to: [140609252](https://github.com/OData/odata.net/pull/960#discussion_r140609252) [](ancestors = 140609252)